### DISCUSSION TITLE
grammar.js use many undefined functions which bring many warnings and errors. How to avoid them?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [tree-sitter/tree-sitter#2801] [Q&A] [answered]
![Screenshot from 2023-11-30 14-09-12](https://github.com/tree-sitter/tree-sitter/assets/32936898/cad01fe3-d427-4982-adb8-a6eb80fee252)

**Comment by amaanq** [comment, marked as answer]
Everything's defined here - https://github.com/tree-sitter/tree-sitter/blob/0ff28346be3d27f935d7cde8bbdf6b621c268e1a/cli/npm/dsl.d.ts

**Reply by Freed-Wu** [reply]
Right, add

```js
/// <reference types="tree-sitter-cli/dsl" />
```

can make it work.
